harpy/bin/deconvolve_alignments.py

```
@@ -0,0 +1,204 @@
+# LAST_CONTIG keeps track of the last contig so we can
+# clear the dict when it's a new contig/chromosome
+LAST_CONTIG = False
+# MI is the name of the current molecule, starting a 1 (0+1)
+MI = 0
+
+if not os.path.exists(bam_input):
+    sys.stderr.write(f"Error: {bam_input} not found\n")
+    sys.exit(1)
+
+if bam_input.lower().endswith(".bam"):
+    if not os.path.exists(bam_input + ".bai"):
```

Ruff

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)